src/main/java/org/assertj/core/extention/SoftlyExtension.java

```diff
@@ -0,0 +1,112 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.extention;
+
+import org.assertj.core.annotations.Softly;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.internal.Failures;
+import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.TestInstancePostProcessor;
+import org.junit.platform.commons.support.AnnotationSupport;
+
+import java.lang.reflect.Field;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+
+/**
+ * <p>
+ * Injects an instance of {@link SoftAssertions} to any Field of type
+ * {@code SoftAssertions} that has the annotation {@link Softly}
+ * before each test method execution and invokes
+ * {@link SoftAssertions#assertAll()} after the test method
+ * </p>
+ * <p>
+ * This extension will throw a {@link Failures#failure(String)}
+ * when more that one filed is annotated with{@code @Softly}.
+ * </p>
+ * <p>
+ * A nested test class can provide a {@code SoftAssertions}
+ * field when it extends {@code this} extension or can inherit
+ * the parent's {@code Soft assertions field}</p>
+ * <p>
+ * Inspired by Mockito @Captor annotation.
+ * </p>
+ *
+ * @author Arthur Mita
+ **/
+public class SoftlyExtension implements TestInstancePostProcessor, BeforeEachCallback, AfterTestExecutionCallback {
+
+  private static final Logger LOGGER = Logger.getLogger(SoftlyExtension.class.getName());
+  private static final Namespace ASSERTJ_CORE = Namespace.create("org.assertj.core");
+  private static final String TEST_INSTANCE = "testInstance";
+  private final Set<SoftAssertions> softlies = new HashSet<>();
+
+  private Failures failures = Failures.instance();
+
+  @Override
+  public void postProcessTestInstance(Object testInstance, ExtensionContext extensionContext) throws Exception {
+    extensionContext.getStore(ASSERTJ_CORE).put(TEST_INSTANCE, testInstance);
+  }
+
+  @Override
+  public void beforeEach(ExtensionContext extensionContext) throws Exception {
+    Optional<ExtensionContext> currentContext = Optional.of(extensionContext);
+
+    while (currentContext.isPresent()) {
+      ExtensionContext context = currentContext.get();
+      Object testInstance = context.getStore(ASSERTJ_CORE).remove(TEST_INSTANCE);
```

Comment:
why removing it instead of simply getting it? is there some concurrency issue I'm not seeing?

Reply:
I figured we do not need the instance since we have already had the `SoftAssertions` instance set. There was no concurrency issue that I could detect.

Reply:
could we use https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/extension/ExtensionContext.html#getTestInstance() instead of accessing the store?

Reply:
If I understand correctly context are per tests (can access the parent context if needed), so in BeforeTestExecutionCallback we could simply store in the built soft assertion in the current ExtensionContext  and use it in afterTestExecution, WDYT? 
execution

Reply:
Yes I think that would be cleaner. I think the store should be used to store objects that need to be accessed across multiple test instances which is not our current use case. Let me work on this.